CommonControls/Assimp/AssimpImporter.cs

```diff
@@ -0,0 +1,350 @@
+                var v1 = assInputMesh.Faces[assFaceIndex].Indices[1];
+                var v2 = assInputMesh.Faces[assFaceIndex].Indices[2];
+
+                unindexesMesh.IndexList[assFaceIndex * 3 + 0] = (ushort)v0;
+                unindexesMesh.IndexList[assFaceIndex * 3 + 1] = (ushort)v1;
+                unindexesMesh.IndexList[assFaceIndex * 3 + 2] = (ushort)v2;
+            }
+
+            return unindexesMesh;
+        }
+
+        /// <summary>
+        ///  Make packed vertices from the assimp meshs attributes
+        /// </summary>        
+        private List<CommonVertex> MakePackedVertices(Assimp.Mesh assInputMesh)
+        {
+            List<CommonVertex> vertices = new CommonVertex[assInputMesh.VertexCount].ToList();
+
+            for (int vertexIndex = 0; vertexIndex < vertices.Count; vertexIndex++)
+            {
+                vertices[vertexIndex] = MakePackedVertex(assInputMesh.Vertices[vertexIndex], assInputMesh.Normals[vertexIndex], assInputMesh.TextureCoordinateChannels[0][vertexIndex], assInputMesh.Tangents[vertexIndex], assInputMesh.BiTangents[vertexIndex]);
+            }
+
+            if (_skeletonFile != null)
+            {
+                ProcessWeights(vertices, assInputMesh);
+            }
+
+            return vertices;
+        }
+
+        /// <summary>
+        ///  Make packed vertex from position, normal and UV
+        /// </summary>        
+        private CommonVertex MakePackedVertex(
+            Assimp.Vector3D position,
+            Assimp.Vector3D normal,
+            Assimp.Vector3D textureCoords,
+            Assimp.Vector3D tangent,
+            Assimp.Vector3D bitangent)
+        {
+            // -- Attempt at getting unit values from the "native"
+            //Scene.Metadata.TryGetValue("UnitScaleFactor", out var value);
+            //float scaleFactor = 1 / (float)value.DataAs<double>();
+
+            // -- Assimp outpus unnormalized normals and tangents.
+            var normalizedNormal = new Vector3(-normal.X, normal.Y, normal.Z);
+            normalizedNormal.Normalize();
+
+            var normalizedTangent = new Vector3(-tangent.X, tangent.Y, tangent.Z);
+            normalizedTangent.Normalize();
+
+            var normalizedBitangent = new Vector3(-bitangent.X, bitangent.Y, bitangent.Z);
+            normalizedBitangent.Normalize();
+
+            var vertex = new CommonVertex()
+            {
+                Position = new Vector4(-position.X, position.Y, position.Z, 0),// * scaleFactor,
+                Normal = normalizedNormal,
+                Uv = new Vector2(textureCoords.X, -textureCoords.Y),
+                //Uv = new Vector2(textureCoords.X, textureCoords.Y),
+                Tangent = normalizedTangent,
+                BiNormal = normalizedBitangent,
+                WeightCount = 0,
+            };
+
+            if (_skeletonFile != null)
+            {
+                vertex.BoneIndex = new byte[4];
```

Comment:
Consider assigin a variable inside the if and then using it to make the weights
var numWeights = 4;
if(skeleton== null)
numWeights = 0;
  vertex.BoneIndex = new byte[numWeights ];
                vertex.BoneWeight = new float[numWeights ];